src/request/request.ts

```diff
@@ -1,87 +1,91 @@
 import { MindlessError } from '../error/mindless.error'
-import { Event, HttpMethods } from './event'
-import { IRequest } from './request-interface'
-
-export class Request implements IRequest {
-  public RouteMetaData: { [key: string]: any } = {}
-  protected _body: { [key: string]: any }
-  protected data: { [key: string]: any } = {}
-
-  constructor(protected event: Event) {
-    if (event.body === '' || event.body == null) {
-      this._body = {}
-    } else {
-      this._body = JSON.parse(event.body)
-    }
-    if (this.event.pathParameters == null) {
-      this.event.pathParameters = {}
-    }
-    if (this.event.queryStringParameters == null) {
-      this.event.queryStringParameters = {}
-    }
-    if (this.event.headers == null) {
-      this.event.headers = {}
-    }
+import { RouteMetadata } from '../routing/IRouter'
+import { HttpMethods } from './http-methods'
+
+export class Request<TBody = { [key: string]: any }> {
+  protected _pathParam: Readonly<{ [key: string]: string }>
```

Comment:
Think we should make all these maps?

Reply:
what is the value? It is a little bit of overhead (need to iterate through the whole object

Reply:
If we update our request event and request data to use maps it forces the users of this library to do the translation and I think I am fine with that 

we would then be using es6 maps throughout which i like]

however, i think it is out of the scope of this PR and I will open an issue to address in the future

Reply:
on second thought it is completely within the scope of the PR lol